File: src/read/file_read.py
import pathlib

import pandas as pd
import yaml

from src.path import pathOfDataFile


# read BAT CSV input file
def readTEDFile(path: pathlib.Path, mapColnamesDtypes: dict):
    return pd.read_csv(path, names=list(mapColnamesDtypes.keys()), dtype=mapColnamesDtypes, encoding='utf-8')


# read CSV data file
def readCSVDataFile(fname: str):
    path = pathOfDataFile(fname)
    return pd.read_csv(path)


# read YAML config file
def readYAMLDataFile(fname: str):
    path = pathOfDataFile(f"{fname}.yml")
    return yaml.load(open(path, 'r', encoding='utf-8').read(), Loader=yaml.FullLoader)

File: src/R/read/file_read.R
library(yaml)

source("src/R/path.R")


# read TED CSV input file
readTEDFile <- function (path, mapColnamesDtypes){
    return(read.csv(
      path,
      col.names=list(mapColnamesDtypes.keys()),
      dtype=mapColnamesDtypes,
      sep=',',
      quote='"',
      encoding='utf-8'
    ))
}


# read CSV data file
readCSVDataFile <- function (fname) {
    fpath <- pathOfDataFile(fname)
    return(read.csv(
      fpath,
      sep=',',
      quote='"',
      encoding='utf-8'
    ))
}


# read YAML config file
readYAMLDataFile <- function (fname) {
    fpath <- pathOfDataFile(paste0(fname, ".yml"))
    fhandle <- file(fpath, "r")
    ret <- read_yaml(file=fhandle, fileEncoding="utf-8")
    close(fhandle)
    return(ret)
}
